sdk/src/record-scanner.ts

```diff
@@ -0,0 +1,362 @@
+            throw error;
+        }
+    }
+
+    /**
+     * Find records in the record scanner service.
+     * 
+     * @param {OwnedFilter} filter The filter to use to find the records.
+     * @returns {Promise<OwnedRecord[]>} The records.
+     */
+    async findRecords(filter: OwnedFilter): Promise<OwnedRecord[]> {
+        if (!this.uuid) {
+            throw new Error("Not registered");
+        }
+
+        filter.uuid = this.uuid;
+
+        try {
+            const response = await this.request(
+                new Request(`${this.url}/records/owned`, {
+                    method: "POST",
+                    headers: { "Content-Type": "application/json" },
+                    body: JSON.stringify(filter),
+                }),
+            );
+
+            return await response.json();
+        } catch (error) {
+            console.error(`Failed to get owned records: ${error}`);
+            throw error;
+        }
+    }
+
+    /**
+     * Find a credits record in the record scanner service.
+     * 
+     * @param {number} microcredits The amount of microcredits to find.
+     * @param {OwnedFilter} searchParameters The filter to use to find the record.
+     * @returns {Promise<OwnedRecord>} The record.
+     */
+    async findCreditsRecord(microcredits: number, searchParameters: OwnedFilter): Promise<OwnedRecord> {
+        try {
+            const records = await this.findRecords({
+                decrypt: true,
+                unspent: searchParameters.unspent ?? false,
+                filter: {
+                    start: searchParameters.filter?.start ?? 0,
+                    program: "credits.aleo",
+                    record: "credits",
+                },
+                uuid: this.uuid,
+            });
+
+            const record = records.find(record => {
+                const plaintext = RecordPlaintext.fromString(record.record_plaintext ?? '');
+                const amountStr = plaintext.getMember("microcredits").toString();
+                const amount = parseInt(amountStr.replace("u64", ""));
+                return amount >= microcredits;
+            });
+
+            if (!record) {
+                throw new Error("Record not found");
+            }
+
+            return record;
+        } catch (error) {
+            console.error(`Failed to find credits record: ${error}`);
+            throw error;
+        }
+    }
+
+    /**
+     * Find credits records using a record scanning service.
+     * 
+     * @param {number[]} microcreditAmounts The amounts of microcredits to find.
+     * @param {OwnedFilter} searchParameters The filter to use to find the records.
+     * @returns {Promise<OwnedRecord[]>} The records
+     */
+    async findCreditsRecords(microcreditAmounts: number[], searchParameters: OwnedFilter): Promise<OwnedRecord[]> {
+        try {
+            const records = await this.findRecords({
+                decrypt: true,
+                unspent: searchParameters.unspent ?? false,
```

Comment:
Ditto here if we don't find the unspent flag, we shouldn't set it to false as default because that'll return only `spent` records. We just want to return spent OR unspent records in this case, and we just don't specify the flag if that's the case.